tchannel/schemes/thrift.py

```diff
@@ -0,0 +1,45 @@
+from __future__ import (
+    absolute_import, division, print_function, unicode_literals
+)
+
+from tornado import gen
+
+from . import THRIFT
+from tchannel.thrift.scheme import ThriftArgScheme as DepThriftSerializer
+
+
+class ThriftArgScheme(object):
+
+    NAME = THRIFT
+
+    def __init__(self, tchannel):
+        self._tchannel = tchannel
+
+    @gen.coroutine
+    def __call__(self, request=None, header=None, timeout=None, hostport=None):
+
+        # serialize
+        serializer = DepThriftSerializer(request.result_type)
+        header = serializer.serialize_header(header)
+        body = serializer.serialize_body(request.call_args)
+
+        response = yield self._tchannel.call(
+            scheme=self.NAME,
+            service=request.service,
+            arg1=request.endpoint,
+            arg2=header,
+            arg3=body,
+            hostport=hostport
```

Comment:
Don't forget to remove the distinction between service & hostport if they are never used in practice simultaneously.